Also failure with main branch too: eb0c485b6c836abb71932537a5058344d11d7bc8

````
======================================================================
FAIL: test_singlephase_variants (test.test_imp.ImportTests.test_singlephase_variants) [_testsinglephase]
Exercise the most meaningful variants described in Python/import.c.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/cpython/Lib/test/test_imp.py", line 366, in test_singlephase_variants
    self.assertEqual(init_count, expected_init_count)
AssertionError: 3 != 1

======================================================================
FAIL: test_singlephase_variants (test.test_imp.ImportTests.test_singlephase_variants) [_testsinglephase_basic_wrapper]
Exercise the most meaningful variants described in Python/import.c.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/cpython/Lib/test/test_imp.py", line 383, in test_singlephase_variants
    self.assertEqual(init_count, expected_init_count)
AssertionError: 3 != 2

----------------------------------------------------------------------
Ran 30 tests in 0.106s
````

--- a/Lib/importlib/_bootstrap.py
+++ b/Lib/importlib/_bootstrap.py
@@ -89,6 +89,7 @@ class _BlockingOnManager:
             # gh-101766: glboal cache should be cleaned-up
             # if there is no more _blocking_on for this thread.
             del _blocking_on[self.thread_id]
+            del self.blocked_on
 
 
 class _DeadlockError(RuntimeError):

--- a/Lib/importlib/_bootstrap.py
+++ b/Lib/importlib/_bootstrap.py
@@ -85,6 +85,10 @@ class _BlockingOnManager:
     def __exit__(self, *args, **kwargs):
         """Remove self.lock from this thread's _blocking_on list."""
         self.blocked_on.remove(self.lock)
+        if len(self.blocked_on) == 0:
+            # gh-101766: glboal cache should be cleaned-up
+            # if there is no more _blocking_on for this thread.
+            del _blocking_on[self.thread_id]
 
 
 class _DeadlockError(RuntimeError):

